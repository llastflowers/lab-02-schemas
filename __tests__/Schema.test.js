const { Schema } = require('../lib/Schema.js');

describe('Schema', () => {
    it('validates a valid schema', () => {
        const schema = new Schema({
            name: {
                type: String,
                required: true
            },
            age: {
                type: Number
            },
            color: {
                type: String
            }
        });

        const cat = {
            name: 'willow',
            age: 3,
            color: 'black and white'
        };

        expect(schema.validate(cat)).toEqual({
            name: 'willow',
            age: 3,
            color: 'black and white'
        });
    });

    it('throws an error if a schema is invalid', () => {
        const schema = new Schema({
            name: {
                type: String,
                required: true
            },
            age: {
                type: Number
            },
            weight: {
                type: String
            }
        });

        const cat = {
            age: 'yes',
            color: 'black and white'
        };

        expect(() => schema.validate(cat)).toThrowErrorMatchingSnapshot();
    });
});