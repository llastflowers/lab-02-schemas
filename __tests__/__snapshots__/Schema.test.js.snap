// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema throws an error if a schema is invalid 1`] = `"invalid schema >> Error: Required field name not found,Error: Cannot cast >>yes<< to Number"`;
